subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+            "property": "subject",
+            "value": ".jpg"
+        },
+    ]
+}
+```
+
+#### 3.2.4. API Operations
+
+This section enumerates the abstract operations that are defined for
+subscription managers. The following sections define bindings of these abstract
+operations to concrete protocols.
+
+The operations are `Create`, `Retrieve`, `Query`, `Update`, and `Delete`. Of
+those, only the `Retrieve` operation is REQUIRED for conformance. The `Create`
+and `Delete` operations SHOULD be implemented. `Query` and `Update` are
+OPTIONAL.
+
+Protocol bindings SHOULD provide a discovery mechanism for which operations are
+supported.
+
+#### 3.2.4.1. Creating a subscription
+
+The **Create** operation SHOULD be supported by compliant Event Producers. It
+creates a new Subscription. The client proposes a subscription object which MUST
+contain all REQUIRED properties. The subscription manager then realizes the
+subscription and returns a subscription object that also contains all OPTIONAL
+properties for which default values have been applied.
+
+Parameters:
+
+- subscription (subscription) - REQUIRED. Proposed subscription object.
+
+Result:
+
+- subscription (subscription) - REQUIRED. Realized subscription object.
+
+Errors:
+
+- **ok** - the operation succeeded
+- **conflict** - a subscription with the given _id_ already exists
+- **invalid** - the proposed subscription object contains invalid information
+
+Protocol bindings MAY map the Create operation such that the proposed _id_ is
+ignored and the subscription manager assigns one instead.
+
+#### 3.2.4.2. Retrieving a Subscription
+
+The **Retrieve** operation MUST be supported by compliant Event Producers. It
+returns the specification of the identified subscription.
+
+Parameters:
+
+- id (string) - REQUIRED. Identifier of the subscription.
+
+Result:
+
+- subscription (subscription) - REQUIRED. Subscription object.
+
+Errors:
+
+- **ok** - the operation succeeded
+- **notfound** - a subscription with the given _id_ already exists
+
+#### 3.2.4.3. Querying for a list of Subscriptions
+
+The **Query** operation SHOULD be supported by compliant Event Producers. It
+allows to query the list of subscriptions on the subscription manager associated
+with or otherwise visible to the party making the request. If supported, it MUST
+be supported at the same endpoint as the **Create** subscription operation.
+
+Parameters:
+
+- none
+
+Result:
+
+- subscription (list of subscription) - REQUIRED. List of subscription objects
+
+Errors:
+
+- **ok** - the operation succeeded and returned results
+- **nocontent** - the operation succeeded and returned no results
+
+Protocol bindings and implementations of such bindings MAY add custom filter
+constraints and pagination arguments as parameters. A request without filtering
+constraints SHOULD return all available subscriptions associated with or
+otherwise visible to the party making the request.
+
+#### 3.2.4.4. Updating a Subscription
```

Comment:
If a subscription’s filter gets updated, it might mean that the consumer have now previously received a subset of the messages described by the subscription (especially when the subscription manager stores the history of events). Is it worth taking on stance on the fact that we don’t expect consumers to receive historical events that have not been sent to it do to a previously “smaller” filter?

Reply:
My 2 cents is that I think this is going to be technology and implementation dependent and should be defined in the binding for the protocol.

Reply:
@sroze Do you know a broker that lets you go back in time in this fashion?